packages/nimbus/src/components/data-table/components/data-table.manager.tsx

```diff
@@ -0,0 +1,286 @@
+      }
+    });
+
+    setVisibleColumns(resetVisible);
+    setHiddenColumns(resetHidden);
+
+    // Notify parent of reset
+    if (onColumnsChange) {
+      onColumnsChange(allColumns);
+    }
+  };
+
+  const handleVisibleColumnsReorder = (items: ColumnListItem[]) => {
+    setVisibleColumns(items);
+
+    // Notify parent of reorder
+    if (onColumnsChange) {
+      const reorderedColumns = items.map((item) => item.column);
+      const hiddenColumnsData = hiddenColumns.map((item) => item.column);
+      onColumnsChange([...reorderedColumns, ...hiddenColumnsData]);
+    }
+  };
+
+  const defaultTrigger = (
+    <IconButton
+      variant="ghost"
+      tone="primary"
+      size="md"
+      aria-label={formatMessage(messages.settings)}
+      onClick={() => setIsOpen(true)}
+    >
+      <Settings />
+    </IconButton>
+  );
+
+  return (
+    <>
+      {renderTrigger
+        ? renderTrigger({ onClick: () => setIsOpen(true) })
+        : defaultTrigger}
+
+      <Drawer.Root isOpen={isOpen} onOpenChange={setIsOpen} placement="right">
+        <Drawer.Content width="640px">
+          <Drawer.Header>
+            <Drawer.Title>{formatMessage(messages.settings)}</Drawer.Title>
+            <Drawer.CloseTrigger />
+          </Drawer.Header>
+
+          <Drawer.Body>
+            <Tabs.Root
+              tabs={[
+                {
+                  id: "visible-columns",
+                  tabLabel: (
+                    <>
+                      <ViewWeek />
+                      {formatMessage(messages.visibleColumns)}
+                    </>
+                  ),
+                  panelContent: (
+                    <Stack gap="400" mt="400">
+                      <Flex gap="400" width="100%" mb="800">
+                        {/* Hidden Columns Section */}
+                        <Box>
+                          <Stack direction="row" alignItems="center" mb="200">
+                            <VisibilityOff />
+                            <Text fontWeight="700" fontSize="sm">
+                              {formatMessage(messages.hiddenColumns)}
+                            </Text>
+                          </Stack>
+                          <DraggableList.Root
+                            h="full"
+                            items={hiddenColumns.map(
+                              (item: ColumnListItem) => ({
+                                key: item.key,
+                                id: item.id,
+                                label: item.label,
+                              })
+                            )}
+                            onUpdateItems={(items) =>
+                              handleVisibleColumnsReorder(
+                                items as ColumnListItem[]
+                              )
+                            }
+                            aria-label={formatMessage(
+                              messages.hiddenColumnsAriaLabel
+                            )}
```

Comment:
an `aria-label` isn't necessary if you use the `FormField` to associate a label with the list